test/runtests.jl

```diff
@@ -1,51 +1,50 @@
+### Fecth the require packages ###
 using SafeTestsets
 
+
+### Decalre constants realted to groups.
 const GROUP = get(ENV, "GROUP", "All")
 const is_APPVEYOR = Sys.iswindows() && haskey(ENV,"APPVEYOR")
 const is_TRAVIS = haskey(ENV,"TRAVIS")
 
-# full macro tests
+
+### Ruyn the tests ###
 @time begin
 
-if GROUP == "All" || GROUP == "Core"
-  @time @safetestset "Model Macro" begin include("make_model_test.jl") end
-  @time @safetestset "Gillespie Tests" begin include("gillespie.jl") end
-  @time @safetestset "Test Solvers" begin include("solver_test.jl") end
+# Tests all features realted to constructing a model
+if GROUP == "All" || GROUP == "Network Creation & Modification"
+  @time @safetestset "Model Construction" begin include("make_model.jl") end
+  @time @safetestset "Custom Functions" begin include("custom_functions.jl") end
+  #@time @safetestset "Model Modification" begin include("model_modification.jl") end
+end
+
+# Tests various core properties of the package.
+if GROUP == "All" || GROUP == "Core Properties"
   @time @safetestset "Higher Order" begin include("higher_order_reactions.jl") end
-  @time @safetestset "Additional Functions" begin include("func_test.jl") end
-  @time @safetestset "Steady State Solver" begin include("steady_state.jl") end
-  @time @safetestset "Mass Action Jumps" begin include("mass_act_jump_tests.jl") end
-  @time @safetestset "Other Tests" begin include("misc_tests.jl") end
-  @time @safetestset "Network query tests" begin include("networkquery_test.jl") end
+  @time @safetestset "U0 and Parameters Input Types" begin include("u0_n_parameter_inputs.jl") end
+  #@time @safetestset "Network Property Queries" begin include("property_query.jl") end
 end
 
-if !is_APPVEYOR && (GROUP == "All" || GROUP == "EquibrilateI")
-  @time @safetestset "Equilibrate (1)" begin include("equilibrate_test_1.jl") end
-  @time @safetestset "Equilibrate (2)" begin include("equilibrate_test_2.jl") end
+# Tests related to solving Ordinary Differential Equations.
+if GROUP == "All" || GROUP == "Deterministic Tests"
+  @time @safetestset "ODE System Solving" begin include("solve_ODEs.jl") end
+  @time @safetestset "Make Jacobian" begin include("make_jacobian.jl") end
 end
 
-if !is_APPVEYOR && (GROUP == "All" || GROUP == "EquibrilateII")
-  @time @safetestset "Equilibrate (3)" begin include("equilibrate_test_3.jl") end
-  @time @safetestset "Equilibrate (4)" begin include("equilibrate_test_4.jl") end
+# Tests related to solving Stochastic Differential Equations.
+if GROUP == "All" || GROUP == "SDE Problems"
+  @time @safetestset "SDE System Solving" begin include("solve_SDEs.jl") end
```

Comment:
is this a major regression? Or was it taking like 30 minutes before?

Reply:
Not sure what you mean? I scratched the equi tests since that feature is temporarily removed.

For the splitting SDE Problems I just divided things into a couple of natural groups. However, admittedly I don't really know what is the preferred way to group things! So things might seem weird to someone actually understanding it. Feel more than happy to make modifications as you seem fit.

None of the current tests should take anywhere close to 30min. Some of the solvers are not fast, but it would be very easy to remove a couple of repeats (e.g. at one point I create and run an ODEProblem over a single time unit, for 44 different networks, and does so 3 times each for random `u0`/`p`. This is probably overdoing it...)

Reply:
Oh I see. We usually split to hit timings on Travis, not "natural groups". The more you split the more you have to pay compile times. Normally this is only done to keep tests below 60 minutes.

Reply:
I see, yeah, we don't have to worry about that kind of run times! I will run some timing tests and then re-group.

Reply:
IIRC they were already split at 30-40 minute groups.